### PR TITLE
fix: add the xy-grid parameter to foundation-everything

### PR DESCRIPTION
**Pull request by DanielRuf**
This adds `$xy-grid` to `foundation-everything`.

Closes https://github.com/zurb/foundation-sites/issues/10742

### CODE REVIEW

**Review by ncoden — CHANGES_REQUESTED** [review]
This makes the global `$xy-grid` config ([scss/xy-grid/_xy-grid.scss:11](https://github.com/zurb/foundation-sites/blob/f29e2d2ecf818844e62e7355846991f8b2989145/scss/xy-grid/_xy-grid.scss#L11)) useless.

We have the following solutions:
* use the global `$xy-grid` by default as parameter (better)
* set the global `$xy-grid` with the parameter (consistent with the `$flex` parameter)

I would personnally go for the _consistent_ option for now, but I'd like to get your opinion.

**Comment by DanielRuf** [comment]
We have no `$global-xy-grid`, so how should we name the variable here in this file?

**Comment by ncoden** [comment]
IMHO `$xy-grid`should work here as a param _and_ a global config.

**Comment by ncoden** [comment]
@if $xy-grid { // test the local parameter
    $xy-grid: true !global; // set the global config
}

**Comment by DanielRuf** [comment]
> @if $xy-grid { // test the local parameter
>     $xy-grid: true !global; // set the global config
> }

Changed it according to your suggestion.

File: scss/foundation.scss [hunk 1]
```diff
@@ -69,8 +69,13 @@
 
 @mixin foundation-everything(
   $flex: true,
-  $prototype: false
+  $prototype: false,
+  $xy-grid: true
 ) {
+  @if $xy-grid {
+    $xy-grid: true !global;
+  }
```
Comment (ncoden) [inline scss/foundation.scss]:
```scss
$prototype: false,
$xy-grid: true
```
So:
```scss
@if $flex { 
  $global-flexbox: true !global; 
} 
@if $xy-grid { 
  $xy-grid: true !global; 
} 
```

Does it works as expected ? (changing the global variable).

File: scss/foundation.scss [hunk 2]
```diff
@@ -69,8 +69,13 @@
 
 @mixin foundation-everything(
   $flex: true,
-  $prototype: false
+  $prototype: false,
+  $xy-grid: true
```
Comment (ncoden) [inline scss/foundation.scss]: In this case I think we should use:
```scss
$xy-grid: $xy-grid
```
for backward compatibility. The global `$xy-grid` setting should still have an effect.
Please check it works as expected.

**Review by ncoden — CHANGES_REQUESTED** [review]

File: scss/foundation.scss (hunk 2 above)
Comment (DanielRuf) [reply scss/foundation.scss]: Ok, will check this later today before pushing this proposed change.

**Comment by DanielRuf** [comment]
Is something left or did we oversee something?

**Comment by DanielRuf** [comment]
> style: change the code formatting without effects on the execution

Would be this then.

**Pull request merged**